main.go

```diff
@@ -3,208 +3,257 @@ package main
 import (
 	"flag"
 	"fmt"
+	"io/ioutil"
 	"net/http"
 	"strconv"
-	"sync"
+	"strings"
+	"time"
 
 	"github.com/Snapbug/gomemcache/memcache"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/log"
 )
 
 const (
-	namespace = "memcache"
+	namespace = "memcached"
 )
 
 var (
 	Version = "0.0.0"
-
-	cacheOperations  = []string{"get", "delete", "incr", "decr", "cas", "touch"}
-	cacheStatuses    = []string{"hits", "misses"}
-	usageTimes       = []string{"curr", "total"}
-	usageResources   = []string{"items", "connections"}
-	bytesDirections  = []string{"read", "written"}
-	removalsStatuses = []string{"expired", "evicted"}
 )
 
-// Exporter collects metrics from a set of memcache servers.
+// Exporter collects metrics from a memcached server.
 type Exporter struct {
-	mutex    sync.RWMutex
-	mc       *memcache.Client
-	up       prometheus.Gauge
-	uptime   prometheus.Counter
-	cache    *prometheus.CounterVec
-	usage    *prometheus.GaugeVec
-	bytes    *prometheus.CounterVec
-	removals *prometheus.CounterVec
+	mc *memcache.Client
+
+	up               *prometheus.Desc
+	uptime           *prometheus.Desc
+	version          *prometheus.Desc
+	bytesRead        *prometheus.Desc
+	bytesWritten     *prometheus.Desc
+	connections      *prometheus.Desc
+	connectionsTotal *prometheus.Desc
+	currentBytes     *prometheus.Desc
+	limitBytes       *prometheus.Desc
+	commands         *prometheus.Desc
+	items            *prometheus.Desc
+	itemsTotal       *prometheus.Desc
+	evictions        *prometheus.Desc
+	reclaimed        *prometheus.Desc
 }
 
-// NewExporter returns an initialized exporter
-func NewExporter(server string) *Exporter {
+// NewExporter returns an initialized exporter.
+func NewExporter(server string, timeout time.Duration) *Exporter {
+	c := memcache.New(server)
+	c.Timeout = timeout
+
 	return &Exporter{
-		mc: memcache.New(server),
-		up: prometheus.NewGauge(
-			prometheus.GaugeOpts{
-				Name:        "up",
-				Namespace:   namespace,
-				Help:        "Are the servers up.",
-				ConstLabels: prometheus.Labels{"server": server},
-			},
+		mc: c,
+		up: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "up"),
+			"Could the memcached server be reached.",
+			nil,
+			nil,
+		),
+		uptime: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "uptime"),
+			"The uptime of the server.",
```

Comment:
But those would be about the uptime of the exporter, not the uptime of the memcached server, right?